base/applications/mscutils/eventvwr/evtdetctl.c

```diff
@@ -838,31 +838,26 @@ EventDetailsCtrl(HWND hDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
                 case IDC_NEXT:
                 {
                     BOOL bPrev = (LOWORD(wParam) == IDC_PREVIOUS);
-                    INT iItem, iSel;
+                    INT iItem, iSel, nItems = ListView_GetItemCount(hwndListView);
+                    WCHAR szText[200];
+
+                    if (nItems <= 0) /* No items? */
+                        break;
 
                     /* Select the previous/next item from our current one */
-                    iItem = ListView_GetNextItem(hwndListView,
-                                                 pData->iEventItem,
-                                                 bPrev ? LVNI_ABOVE : LVNI_BELOW);
-                    if (iItem == -1)
+                    iItem = ListView_GetNextItem(hwndListView, -1, LVNI_ALL | LVNI_SELECTED);
```

Comment:
It may be of interest to reset pData->iEventItem to -1, when events are being (re)loaded, if this isn't already done, in the function you've also modified in eventvwr.c 🤔 what do yo think about this?

Reply:
`pData->iEventItem` is used in `EventDetailsCtrl` (details control dialog procedure).
`EventDetailsCtrl` is used by `CreateEventDetailsCtrl` function.
`CreateEventDetailsCtrl` is used in `InitInstance` and `EventDetails` functions.
`InitInstance` is for main window. `EventDetails` is for `IDM_EVENT_DETAILS` message.
If `pData->iEventItem` was not initialized, it is in the case of `lParam == 0`.

Reply:
Fixed.